src/main/scala/terminal/FileSystem.scala

```diff
@@ -0,0 +1,113 @@
+package terminal
+
+import java.util.NoSuchElementException
+
+
+class FileSystem {
+  //key = command;  value = how many paths need to be specified
+  val commandsAndNoOfPaths = Map("pwd" -> 0, "mkdir" -> 1, "ls" -> 0, "h" -> 0, "q" -> 0 )
+
+  var rootPath : Folders = _
+  var currentLevel : Folders = _
+  var cmdPrefix : String = ""
+  var path1_from : String = ""
+  var path2_to : String = ""
+
+
+
+  def run() : Unit = {
+    //initialize the root
+    rootPath = new Folders("")
+    rootPath.currentPath = "/"
+    currentLevel = rootPath
+
+    //run the terminal continuously until is chosen q to quit
+    commandInterpretor()
+  }
+
+
+  def commandInterpretor() : Unit = {
+    print( """Your cmd (press "q" to quit, "h" for help): """ )
+    val keyboardInput = getKeyboardInput
+
+    splitToCmdAndPaths( keyboardInput )
+
+    cmdPrefix match {
+      case "pwd"        => println( getCurrentDirectoryPath ); unitializeCmdAndPaths(); commandInterpretor()
+      case "mkdir"      => if(path1_from.nonEmpty) makeFolder(); unitializeCmdAndPaths(); commandInterpretor()
+      case "ls"         => listAllFilesAndFolders(); unitializeCmdAndPaths(); commandInterpretor()
+      case "h"          => help(); unitializeCmdAndPaths(); commandInterpretor()
+      case "q"          => println("You chose q, the app will close.")
+      case _            => println( "Command not recognized." ); commandInterpretor()     //this default case will keep the "terminal" running even in case if is a command not recognized
+    }
+  }
+
+  def getKeyboardInput : String = scala.io.StdIn.readLine()
+
+  def getCurrentDirectoryPath : String = currentLevel.currentPath
+
+  def help() : Unit = {
```

Comment:
Can we move this method to an utility object?

Reply:
I am not sure if I understood the question, can you please reformulate it?

Reply:
As this method does not use any member/methods from the current class it's in, I suggest to move it to an object that would contain this (and maybe any other methods).

There is also a reason for which it would make sense to have it in this class, since its logic is tight to the class it's part of.

From my point of view, you could pick any approach you want.